org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/adwindow/GridTabRowRenderer.java

```diff
@@ -327,12 +327,22 @@ private Component getDisplayComponent(int rowIndex, Object value, GridField grid
 		} else if (gridField.isHeading()) {
 			component = createInvisibleComponent();
 		} else if (gridField.getDisplayType() == DisplayType.Button) {
+			// Get VO ctx
+			Properties ctx = gridField.getVO().ctx;
+
+			// Each row renderer --- ctx per row wise
 			GridRowCtx gridRowCtx = new GridRowCtx(Env.getCtx(), gridTab, rowIndex);
+			gridField.getVO().setCtx(gridRowCtx);
+
 			WButtonEditor editor = new WButtonEditor(gridField, rowIndex);
 			editor.setValue(gridTab.getValue(rowIndex, gridField.getColumnName()));
 			editor.setReadWrite(gridField.isEditable(gridRowCtx, true,true));
 			editor.getComponent().setAttribute(GRID_ROW_INDEX_ATTR, rowIndex);
 			editor.addActionListener(buttonListener);
+			
+			// Reset origin VO ctx
+			gridField.getVO().setCtx(ctx);
+			//
```

Comment:
why do we need this temporary setting of ctx here ?

Reply:
In this case, creating temporary ctx for the specific row-wise, if not reset origin VO ctx then Panel view button logic does not work as aspected.